forge/test/PoolFactory/DeployPool.t.sol

```diff
@@ -0,0 +1,330 @@
+//SPDX-License-Identifier: CC-BY-NC-ND-4.0
+pragma solidity 0.8.7;
+
+import "../helpers/PoolSetupHelper.t.sol";
+import "interfaces/ILeveragedPool.sol";
+import "testutils/TestClones.sol";
+import "implementation/SMAOracle.sol";
+
+contract PoolFactoryDeployPoolTest is PoolSetupHelper {
+    address constant NON_DAO = ADDR_4;
+
+    uint32 internal frontRunningInterval = 20;
+    uint32 internal updateInterval = 100;
+    uint256 internal fee = 0.1 ether;
+    uint16 internal leverage = 1;
+
+    function setUp() public {
+        setupPoolAndTokenContracts(
+            POOL_CODE,
+            frontRunningInterval,
+            updateInterval,
+            leverage,
+            FEE_RECEIVER,
+            fee,
+            0,
+            0,
+            0
+        );
+    }
+
+    function test_cannot_deploy_when_not_called_by_oracleWrapper_owner_and_with_valid_params() public {
+        IPoolFactory.PoolDeployment memory deploymentParameters = IPoolFactory.PoolDeployment({
+            poolName: POOL_CODE,
+            frontRunningInterval: 5,
+            updateInterval: 10,
+            leverageAmount: 5,
+            settlementToken: address(token),
+            oracleWrapper: address(oracleWrapper),
+            settlementEthOracle: address(settlementEthOracle),
+            feeController: DEPLOYER,
+            mintingFee: 0,
+            burningFee: 0,
+            changeInterval: 0
+        });
+
+        vm.startPrank(NON_DAO);
+        vm.expectRevert("Deployer must be oracle wrapper owner");
+        poolFactory.deployPool(deploymentParameters);
+        vm.stopPrank();
```

Comment:
Is there a way to call a function from a particular `address`, without requiring the use of pranks? Like just `poolFactory.connect(NON_DAO).deployPool` or something of that sort.

Reply:
Not that I'm aware of